### ISSUE TITLE
RFE: Syntax sugar for bconds with inherited defaults

### ISSUE THREAD

**Issue by hroncok** [rpm-software-management/rpm#941]
I often find myself writing code like this:

```
%bcond_with some_thing

%if %{with some_thing}
%bcond_with some_other_thing
%else
%bcond_without some_other_thing
%endif
```

Or the other way around:

```
%bcond_with some_thing

%if %{with some_thing}
%bcond_without some_other_thing
%else
%bcond_with some_other_thing
%endif
```

This allows to do reasonable defaults based on other defaults:

 - if boostrapping, disable tests and documentation build
 - if compatibility package, disallow the devel subpackage

While preserving the ability to customize the bahavior using `--with`/ `--without`:

- disable tests even when not bootstrapping
- build the compatibility package with the devel subpackage

I'd like to have some kind of syntax sugar for this, such as for example:

```
%bcond_with some_thing
%bcond_inherit some_other_thing some_thing
# or
%bcond_antiinherit some_other_thing some_thing
```

Or:

```
%bcond_with some_thing
%bcond_some_thing_inherit some_other_thing
# or
%bcond_some_thing_antiinherit some_other_thing
```
Would that be desirable? I can work on a PR if we agree on the syntax.

**Comment by torsava** [comment]
I would find it useful as well. However, we should really consider the naming, as bconds are already confusing enough, this could become completely unreadable to people not-in-the-know.

**Comment by hroncok** [comment]
I agree. `%bcond_inherit` and `%bcond_antiinherit` are indeed not very good. Brainstorming some ideas:

 * bcond_default_as
 * bcond_default_like
 * bcond_like
 * bcond_as

For the opposite relation, I'm unsure. Mayeb we can use something like

```
%bcond_like ! bootstrap
```

However I'm unsure if `!` can be handled as macro argument.

**Comment by torsava** [comment]
Brainstorming:

* bcond_default_same + bcond_default_opposite

**Comment by encukou** [comment]
My old suggestion was `%bcond <name> <default>`, with numeric "default" (zero for false).
Aside from "inheritance", it might be used instead of `%bcond_with`/`%bcond_without`. (Whenever I try to choose one, I have to stop and think which one will give me the right default. I don't think I'm alone in that.)
```
%bcond some_thing 0
%bcond some_inherited_thing %{with some_thing}
%bcond some_antiinherited_thing %{without some_thing}
```
